images/virtualization-artifact/pkg/controller/vi/internal/storageclass_ready.go

```diff
@@ -129,6 +137,23 @@ func (h StorageClassReadyHandler) setFromSpec(ctx context.Context, vi *virtv2.Vi
 		return fmt.Errorf("get storage class specified in spec: %w", err)
 	}
 
+	if sc != nil {
+		sp, err := h.svc.GetStorageProfile(ctx, sc.Name)
+		if err != nil {
```

Comment:
The code inside this `if` is a copy of validateStorageClass. Is it intentional or is it possible to just call this method here?